assignment01-voronoi/worldgen-voronoi.c

```diff
@@ -0,0 +1,224 @@
+#define BRN "\e[38;5;166m"
+
+void print_grid()
+{
+    for(int i=0; i < GRID_HEIGHT; i++) {
+        for(int j=0; j < GRID_WIDTH; j++) {
+            char* color = "";
+            switch (gridMatrix[i][j]) {
+                case '%':
+                    color = MAG;
+                    break;
+                case '.':
+                    color = YEL;
+                    break;
+                case '~':
+                    color = BLU;
+                    break;
+                case '^':
+                    color = GRN;
+                    break;
+                case ':':
+                    color = GRN;
+                    break;
+	        case '#':
+		    color = BRN;
+		    break;
+                default:
+                    // unreachable 
+                    fprintf(stderr, "unhandled color case\n");
+                    break;
+            }
+            printf("%s%c", color, gridMatrix[i][j]);
+        }
+	putchar('\n');
+    }	       
+}
+
+
+void gen_voronoi_seeds(Seed storedSeeds[SEED_NUM])
+{
+    for(int i=0; i < SEED_NUM; i++){
+        storedSeeds[i].x = 1 + (rand() % (GRID_HEIGHT-1));
+        storedSeeds[i].y = 1 + (rand() % (GRID_WIDTH-1));
+        storedSeeds[i].tile = tiles[i % TileCount]->ascii;
+    }
+    // TODO: above make seed.tile not loop over tiles and have probabilities?
+}
+
+static inline double distance(int x, int y, int x2, int y2)
+{
+    return sqrt( (x-x2) * (x-x2) + (y-y2) * (y-y2));
+}
+
+void voronoi_world_gen(char gridMatrix[GRID_HEIGHT][GRID_WIDTH], Seed storedSeeds[SEED_NUM])
+{   
+    // start at one because of border
+    for (int x =1; x < GRID_HEIGHT - 1 ; x++){
+        for ( int y =1; y < GRID_WIDTH - 1; y++){
+            int closestSeed = 0;
+            double closestDist = distance(x , y, storedSeeds[0].x, storedSeeds[0].y);
+            for(int i=1; i < SEED_NUM; i++){
+                double currDist = distance(x , y, storedSeeds[i].x, storedSeeds[i].y);
+                if(currDist < closestDist){
+                    closestSeed = i;
+                    closestDist = currDist;
+                }
+            }
+            // TODO: this is not desired way to do this but it works
+            // do something similar to printing the colors
+            if(storedSeeds[closestSeed].tile == '^') {
+                int treeChance = rand() % 100;
+                if (treeChance < 25) {
+                    gridMatrix[x][y] = treeTile.ascii;
+                }
+                // else if(25 < treeChance && treeChance < 50) {
+                //     gridMatrix[x][y] = tallGrass.ascii;
+                // }
+                else {
+                    gridMatrix[x][y] = clearing.ascii;
+                }
+            }
+            else {
+                gridMatrix[x][y] = storedSeeds[closestSeed].tile;
+            }
+        }
+     }
+}
+
+void generate_path(char grid[GRID_HEIGHT][GRID_WIDTH], int leftGateX, int leftGateY, int rightGateX, int rightGateY,
```

Comment:
gridMatrix is in the global scope no reason to pass it as param, and an array for gate positions would be better

Reply:
I was going to fix that before i submitted, i realized that as well.